contextproject-ms3/src/main/java/contextproject/audio/MixerProcessor.java

```diff
@@ -0,0 +1,35 @@
+package contextproject.audio;
+
+import be.tarsos.dsp.AudioEvent;
+import be.tarsos.dsp.AudioProcessor;
+
+public class MixerProcessor implements AudioProcessor {
+  private double gain = 1.0f;
+
+  @Override
+  public boolean process(AudioEvent audioEvent) {
+    float[] floatBuffer = audioEvent.getFloatBuffer();
+    for (int i = audioEvent.getOverlap(); i < floatBuffer.length; i++) {
+      float newValue = manipulateSignal(floatBuffer[i]);
+      
+      // Clipping
+      if (newValue > 1.0f) {
+        newValue = 1.0f;
+      } else if (newValue < -1.0f) {
+        newValue = -1.0f;
+      }
+      floatBuffer[i] = newValue;
+    }
+    return true;
+  }
```

Comment:
Note: This entire class is currently unused, but is probably necessary later to combine both audio streams into one and do some limiting/compression, just like a real mixer does.